prov/verbs/src/fi_verbs.c

```diff
@@ -138,7 +138,7 @@ int fi_ibv_create_ep(const char *node, const char *service,
 
 	if (!node && !rai_hints.ai_dst_addr) {
 		if ((!rai_hints.ai_src_addr && !service) ||
-		    (!rai_hints.ai_src_addr &&
+		    (!rai_hints.ai_src_addr && hints &&
```

Comment:
hints->ep_attr would also need to be checked. Ditto below. I was looking at other parts of the code and the check for hints->ep_attr->type == FI_EP_RDM are at several places. I believe all of them should be corrected. It would be better to define a macro to encompass the checks: (hints && hints->ep_attr && (hints->ep_attr == FI_EP_RDM))